ideascube/db_router.py

```diff
@@ -1,30 +1,44 @@
+
+_databaseToUse = {
+    ('search', 'search') : 'transient'
+}
+
 class DatabaseRouter(object):
     """
     A Database router.
     Some models do not need (and must not be backuped).
-    Those models will (must) have a _dbname class attribute set to 'transiant'.
+    So we route them to the transient base.
     This attribute will be use as database name.
     """
     def db_for_read(self, model, **hints):
         """
-        If the model as a _dbname set, use it as db for read
+        Route the table'read to the right base or the default one (None)
         """
-        return getattr(model, '_dbname', None)
+        app_model = model._meta.app_label, model._meta.model_name
+        return _databaseToUse.get(app_model)
 
     def db_for_write(self, model, **hints):
         """
-        If the model as a _dbname set, use it as db for read
+        Route the table'write to the right base or the default one (None)
         """
-        return getattr(model, '_dbname', None)
+        app_model = model._meta.app_label, model._meta.model_name
+        return _databaseToUse.get(app_model)
 
     def allow_relation(self, obj1, obj2, **hints):
         """
         Allow relations if both object are in the same database.
+        This works here cause the only database in transient is the index
+        and it doesn't have 'real' relation.
         """
-        return getattr(obj1, '_dbname', None) == getattr(obj2, '_dbname', None)
+        app_model1 = obj1._meta.app_label, obj1._meta.model_name
+        app_model2 = obj2._meta.app_label, obj2._meta.model_name
+        return _databaseToUse.get(app_model1) == _databaseToUse.get(app_model2)
 
     def allow_migrate(self, db, app_label, model_name=None, **hints):
         """
-        All database need to be migrated.
+        Allow migration only if the migration is applying to the right db.
         """
+        dbToUse = _databaseToUse.get((app_label, model_name), 'default')
+        if dbToUse != db:
+            return False
         return None
```

Comment:
`return None` means "This router takes no decision here".
This could be useful if we add another router to take decision based on another heuristics.

But we have only one router for now and we don't plan to add one in next future, so it's not so important.
As another commit already changes this method to potentially return True, I think it better to actually change the `return None` to `return True`